tests/TestRegistryTest.cpp

```diff
@@ -323,3 +323,38 @@ TEST(TestRegistry, ResetPluginsWorks)
     myRegistry->resetPlugins();
     LONGS_EQUAL(0, myRegistry->countPlugins());
 }
+
+TEST(TestRegistry, listGroupNames)
+{
+    test1->setGroupName("GROUP_1");
+    myRegistry->addTest(test1);
+    test2->setGroupName("GROUP_2");
+    myRegistry->addTest(test2);
+    test3->setGroupName("GROUP_1");
+    myRegistry->addTest(test3);
+
+    myRegistry->listTestGroupNames(*result);
+    SimpleString s = output->getOutput();
+    STRCMP_CONTAINS("GROUP_1 ", s.asCharString());
+    STRCMP_CONTAINS("GROUP_2 ", s.asCharString());
+    LONGS_EQUAL(1, s.count("GROUP_1 "));
```

Comment:
Not really. The test order is quite fixed in this case and you have only one test. If we would decide to change the test order and this test fails, ok then we'll fix it. But now you have 3 'weak' asserts where you can have one 'strong' assert. So, I'd suggest to change it to a stonger assert that asserts the exact output.